File: finalAssignmentBackend/src/main/java/com/tutict/finalassignmentbackend/entity/RequestHistory.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("request_history")
public class RequestHistory implements Serializable {

    @Serial
    private static final long serialVersionUID = 0L;

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField("idempotency_key")
    private String idempotentKey;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField("create_time")
    private LocalDateTime createTime;

    @TableField("business_status")
    private String businessStatus;

    @TableField("business_id")
    private Long businessId;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;
}

File: final_asssignment_backend_go/project/internal/domain/request_history.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// RequestHistory 表示 request_history 表的实体
type RequestHistory struct {
	ID             int64          `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	IdempotentKey  string         `gorm:"column:idempotency_key" json:"idempotency_key"`
	CreateTime     time.Time      `gorm:"column:create_time" json:"create_time"`
	BusinessStatus string         `gorm:"column:business_status" json:"business_status"`
	BusinessID     int64          `gorm:"column:business_id" json:"business_id"`
	DeletedAt      gorm.DeletedAt `gorm:"index"` // 软删除字段（可选，GORM 默认支持）
}

// TableName 指定数据库表名
func (RequestHistory) TableName() string {
	return "request_history"
}
